### ISSUE TITLE
Bug in PrivacyEngine

### ISSUE THREAD

**Issue by shuqike** [awslabs/fast-differential-privacy#16]
The term `self.optimizer` only appears in the function detach() of PrivacyEngine.

That means it leads to AttributeError when you call `privacyengine.detach()`.

So we cannot run the example file [CIFAR_TIMM_2phase.py](https://github.com/awslabs/fast-differential-privacy/blob/main/examples/image_classification/CIFAR_TIMM_2phase.py) properly.

**Comment by woodyx218** [comment]
Fixed. Added one line at L310 https://github.com/awslabs/fast-differential-privacy/blob/main/fastDP/privacy_engine.py
